runtime/src/bank.rs

```diff
@@ -2198,6 +2201,31 @@ impl Bank {
         });
     }
 
+    pub fn update_last_restart_slot(&self) {
+        let feature_flag = self
+            .feature_set
+            .is_active(&feature_set::last_restart_slot_sysvar::id());
+
+        if feature_flag {
+            let last_restart_slot = {
+                let hard_forks = self.hard_forks();
+                let hard_forks_r = hard_forks.read().unwrap();
+                hard_forks_r
+                    .iter()
+                    .last()
+                    .map(|(slot, _)| *slot)
+                    .unwrap_or(0)
+            };
+
+            self.update_sysvar_account(&sysvar::last_restart_slot::id(), |account| {
```

Comment:
@mvines the issue discussed in sdk channel comes from these lines, setting a constant value for last_restart_slot solved the issue.